android/src/main/java/de/motius/RNReactNativeTelephony/RNReactNativeTelephonyModule.java

```diff
@@ -58,12 +58,19 @@ public void getCountryCode(Promise promise){
 
   @ReactMethod
   public void getPhoneNumber(Promise promise){
-    TelephonyManager manager = (TelephonyManager) this.reactContext.getSystemService(Context.TELEPHONY_SERVICE);
-    String phoneNumber = manager.getLine1Number();
-    WritableMap map = Arguments.createMap();
-    map.putString("phoneNumber", phoneNumber);
+    if(getCurrentActivity().checkCallingOrSelfPermission(Manifest.permission.READ_PHONE_STATE) == PackageManager.PERMISSION_GRANTED ||
+            (Build.VERSION.SDK_INT >= 23 && getCurrentActivity().checkCallingOrSelfPermission(Manifest.permission.READ_SMS) == PackageManager.PERMISSION_GRANTED) ||
```

Comment:
I think this `if` statement should be simplified for readability. Extract the three different checks into variables and combine them in the `if`.